core/src/main/java/org/apache/accumulo/core/file/blockfile/cache/lru/CachedBlockQueue.java

```diff
@@ -17,98 +17,89 @@
  */
 package org.apache.accumulo.core.file.blockfile.cache.lru;
 
-import java.util.LinkedList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
 import java.util.PriorityQueue;
 
+import com.google.common.base.Preconditions;
+
 /**
- * A memory-bound queue that will grow until an element brings total size &gt;= maxSize. From then
- * on, only entries that are sorted larger than the smallest current entry will be
- * inserted/replaced.
+ * A memory-bound queue that will grow until an element brings total size &gt;=
+ * maxSize. From then on, only entries that are sorted larger than the smallest
+ * current entry will be inserted and the smallest entry is removed.
  *
  * <p>
- * Use this when you want to find the largest elements (according to their ordering, not their heap
- * size) that consume as close to the specified maxSize as possible. Default behavior is to grow
- * just above rather than just below specified max.
+ * Use this when you want to find the largest elements (according to their
+ * natural ordering, not their heap size) that consume as close to the specified
+ * maxSize as possible. Default behavior is to grow just above rather than just
+ * below specified max.
+ * </p>
  *
  * <p>
- * Object used in this queue must implement {@link HeapSize} as well as {@link Comparable}.
+ * Object used in this queue must implement {@link HeapSize} as well as
+ * {@link Comparable}.
+ * </p>
  */
 public class CachedBlockQueue implements HeapSize {
 
-  private PriorityQueue<CachedBlock> queue;
+  private final PriorityQueue<CachedBlock> queue;
 
+  private final long maxSize;
   private long heapSize;
-  private long maxSize;
 
   /**
-   * @param maxSize
-   *          the target size of elements in the queue
-   * @param blockSize
-   *          expected average size of blocks
+   * Construct a CachedBlockQueue.
+   *
+   * @param maxSize the target size of elements in the queue
+   * @param blockSize expected average size of blocks
    */
-  public CachedBlockQueue(long maxSize, long blockSize) {
-    int initialSize = (int) Math.ceil(maxSize / (double) blockSize);
-    if (initialSize == 0)
-      initialSize++;
-    queue = new PriorityQueue<>(initialSize);
-    heapSize = 0;
+  public CachedBlockQueue(final long maxSize, final long blockSize) {
+    Preconditions.checkArgument(maxSize > 0L);
+    Preconditions.checkArgument(blockSize > 0L);
+
+    final long initialSize = (maxSize + blockSize - 1L) / blockSize;
+
+    this.queue = new PriorityQueue<>(Math.toIntExact(initialSize));
```

Comment:
doesn't toIntExact throw an exception? Should probably test for that an ensure that's either not possible or caught in the call chain.

Reply:
@phrocker Yes.  It does throw an Exception if any data loss occurs from casting from Long to Integer.  That is almost always the desired behavior.  There  would have to be some very funky configuration going on here to create a queue with an initial size of greater than `Integer.MAX_SIZE` and should definitely fail instead of somehow silently managing the overflow.